core/controllers/base.py

```diff
@@ -339,8 +339,9 @@ def render_template(self, filepath, iframe_restriction='DENY'):
         self.response.headers['X-Xss-Protection'] = '1; mode=block'
 
         if iframe_restriction is not None:
-            if iframe_restriction in ['SAMEORIGIN', 'DENY']:
-                self.response.headers['X-Frame-Options'] = iframe_restriction
+            if iframe_restriction in [str('SAMEORIGIN'), str('DENY')]:
```

Comment:
Why these changes?

Reply:
Hi @apb7,
The latest version of web test supports latin 1 string formats while sending a request.
If they are not typecasted into str they where moved to 500 error and the backend tests fails view [here](https://circleci.com/gh/oppia/oppia/9339?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

Reply:
Can you please point me to some documentation for webtest re this change?
Also, you can drop the `str` here since you're already converting `iframe_restriction` to a string below.

Reply:
Hi @apb7,

if we dont str() them, they throw an error `AssertionError: Header names must be latin1 string (not Py2 unicode or Py3 bytes type)`. i've read more about it [here](https://stackoverflow.com/questions/49879770/pyramid-app-header-names-must-be-latin1-string-assertionerror-from-webtest)  and [here](https://www.python.org/dev/peps/pep-3333/#a-note-on-string-types), seems like HTTP response headers should be a native strings - byte strings in Python 2 and unicode strings in Python 3, [here](https://urlzs.com/GhzDg) is the issue raised in webtest repo for the same

Reply:
Can you please point to the changelog of the version which mentions this change? There must be some documentation for this.
And I think you missed this:
> Also, you can drop the str here since you're already converting iframe_restriction to a string below.

I suggest not to blindly put up `str` everywhere without first understanding the need.

Reply:
Hi @apb7 I've made it tedious and justifying please PTAL Thanks!

Reply:
[Here](https://github.com/Pylons/webtest/blob/master/CHANGELOG.rst) is the changelog of wbetest they haven't mentioned in the changelog but the changes in the version below are notable for the cause 
```
2.0.11 : Depend on unittest2 only for Python versions lower than 2.7 [iElectric]
2.0.28 : Fixed #119: Assertion error should be raised when you have non-string response header
```

Also similar notes can be found in requests changelog [here](https://github.com/kennethreitz/requests/blob/master/HISTORY.md) 

`2.14.0 Breaking changes: Much improved handling of non-ASCII Location header values in redirects. Fewer UnicodeDecodeErrors are encountered on Python 2, and Python 3 now correctly understands that Latin-1 is unlikely to be the correct encoding.`

Reading through the above docs and [here](https://legacy.python.org/dev/peps/pep-0333/#unicode-issues) as well looks like the application should return the native strings for headers and Python 2 build-in WSGI server don't help us in encoding/decoding and it must must be handled by the application itself in Python 2 whereas the case is different in Python 3.